sql: TestIndexBackfillerResumePreservesProgress failed

sql.TestIndexBackfillerResumePreservesProgress [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20617464?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20617464?buildTab=artifacts#/) on release-25.4 @ [00809c0b1a2855f926baf7b9f6af9f901b553bd7](https://github.com/cockroachdb/cockroach/commits/00809c0b1a2855f926baf7b9f6af9f901b553bd7):


```
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/18{51/0-61/0}]
    indexbackfiller_test.go:747: pausing backfill
    indexbackfiller_test.go:752: resuming backfill
    indexbackfiller_test.go:613: before resuming backfill, checkpointed spans: [/Table/104/1{-/1861/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/18{61/0-71/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/18{81/0-91/0}]
    indexbackfiller_test.go:747: pausing backfill
    indexbackfiller_test.go:752: resuming backfill
    indexbackfiller_test.go:613: before resuming backfill, checkpointed spans: [/Table/104/1{-/1871/0} /Table/104/1/18{81/0-91/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/18{71/0-81/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/1{891/0-901/0}]
    indexbackfiller_test.go:747: pausing backfill
    indexbackfiller_test.go:752: resuming backfill
    indexbackfiller_test.go:613: before resuming backfill, checkpointed spans: [/Table/104/1{-/1901/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/19{01/0-11/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/19{11/0-21/0}]
    indexbackfiller_test.go:747: pausing backfill
    indexbackfiller_test.go:752: resuming backfill
    indexbackfiller_test.go:613: before resuming backfill, checkpointed spans: [/Table/104/1{-/1921/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/19{21/0-31/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/19{31/0-41/0}]
    indexbackfiller_test.go:747: pausing backfill
    indexbackfiller_test.go:752: resuming backfill
    indexbackfiller_test.go:613: before resuming backfill, checkpointed spans: [/Table/104/1{-/1941/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/19{41/0-51/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/19{51/0-61/0}]
    indexbackfiller_test.go:747: pausing backfill
    indexbackfiller_test.go:752: resuming backfill
    indexbackfiller_test.go:613: before resuming backfill, checkpointed spans: [/Table/104/1{-/1961/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/19{61/0-71/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/19{71/0-81/0}]
    indexbackfiller_test.go:747: pausing backfill
    indexbackfiller_test.go:752: resuming backfill
    indexbackfiller_test.go:613: before resuming backfill, checkpointed spans: [/Table/104/1{-/1981/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/19{81/0-91/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: []
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: []
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/1/{1991/0-2001/0}]
    indexbackfiller_test.go:747: pausing backfill
    indexbackfiller_test.go:752: resuming backfill
    indexbackfiller_test.go:613: before resuming backfill, checkpointed spans: [/Table/104/1{-/2001/0}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: []
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: []
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: []
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: [/Table/104/{1/2001/0-2}]
    indexbackfiller_test.go:605: before index backfill progress update, completed spans: []
    indexbackfiller_test.go:714: condition failed to evaluate within 5s: from indexbackfiller_test.go:736: checkpointed spans [/Table/104/1{-/2001/0}] do not enclose completed spans [/Table/104/{1-2}]
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1061c85d1b78da61f97933da6fe84ebc/logTestIndexBackfillerResumePreservesProgress2783654059
--- FAIL: TestIndexBackfillerResumePreservesProgress (21.78s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154217 sql: TestIndexBackfillerResumePreservesProgress failed [C-test-failure O-robot P-3 T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestIndexBackfillerResumePreservesProgress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55526

--- a/pkg/sql/indexbackfiller_test.go
+++ b/pkg/sql/indexbackfiller_test.go
@@ -668,7 +668,7 @@ func TestIndexBackfillerResumePreservesProgress(t *testing.T) {
 			return err
 		}
 		return nil
-	}, 5*time.Second)
+	}, 30*time.Second)
 
 	ensureJobState := func(targetState string) {
 		testutils.SucceedsWithin(t, func() error {
@@ -682,7 +682,7 @@ func TestIndexBackfillerResumePreservesProgress(t *testing.T) {
 					targetState, jobState)
 			}
 			return nil
-		}, 5*time.Second)
+		}, 30*time.Second)
 	}
 
 	var completedSpans roachpb.SpanGroup
@@ -738,7 +738,7 @@ func TestIndexBackfillerResumePreservesProgress(t *testing.T) {
 			}
 
 			return nil
-		}, 5*time.Second)
+		}, 30*time.Second)
 	}
 
 	for isBlockingBackfillProgress.Load() {

